src/btreemap/iter.rs

```diff
@@ -140,21 +128,82 @@ where
                     },
                     node,
                 });
+            }
+        }
+    }
+
+    fn next_without_loading_value(&mut self) -> Option<()> {
+        // If the cursors are empty. Iteration is complete.
+        let cursor_last = self.cursors.pop()?;
+        self.update_cursors_for_iteration(cursor_last.clone());
+        if let Cursor::Node {
+            node,
+            next: Index::Entry(entry_idx),
+        } = cursor_last
+        {
+            if entry_idx < node.entries_len() {
+                let key = node.key(entry_idx);
 
                 // If the key does not belong to the range, iteration stops.
-                if !self.range.contains(&key) {
+                if !self.range.contains(key) {
                     // Clear all cursors to avoid needless work in subsequent calls.
                     self.cursors = vec![];
                     return None;
                 }
 
-                Some((key, V::from_bytes(Cow::Owned(encoded_value))))
+                return Some(());
             }
-            None => {
-                // The cursors are empty. Iteration is complete.
-                None
+        }
+
+        self.next().map(|_| ())
+    }
+}
+
+impl<K, V, M> Iterator for Iter<'_, K, V, M>
+where
+    K: Storable + Ord + Clone,
+    V: Storable,
+    M: Memory,
+{
+    type Item = (K, V);
+
+    fn next(&mut self) -> Option<Self::Item> {
```

Comment:
This function and `next_without_loading_value` look awfully similar. I wonder if we can somehow combine the two to avoid duplicate logic ;)

Reply:
Done.

Reply:
What do you think about adding a new public fn to Iter `next_key()` which will be almost the same as `next_without_loading_value`? BTreeMap is an ordered set, and providing such functionality without loading values may be useful.

Reply:
`next_key` is not a native function available in `Iterator`, so it cannot be added here IIUC.

Reply:
My idea was that `next_key()` become the function of `Iter` not `Iterator`.

Reply:
Unless there's a strong need for it, I wouldn't worry about it. There are always new things that can be added - better to be as lazy as possible when it comes to these things :)